sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsHostMemoryStore.scala

```diff
@@ -191,6 +216,216 @@ class RapidsHostMemoryStore(
       super.free()
     }
   }
+
+  /**
+   * A per cuDF host column event handler that handles calls to .close()
+   * inside of the `HostColumnVector` lock.
+   */
+  class RapidsHostColumnEventHandler
+    extends HostColumnVector.EventHandler {
+
+    // Every RapidsHostColumnarBatch that references this column has an entry in this map.
+    // The value represents the number of times (normally 1) that a ColumnVector
+    // appears in the RapidsHostColumnarBatch. This is also the HosColumnVector refCount at which
+    // the column is considered spillable.
+    // The map is protected via the ColumnVector lock.
+    private val registration = new mutable.HashMap[RapidsHostColumnarBatch, Int]()
```

Comment:
I think we either need to make this a concurrent hash map or make the methods that modify it synchronized.  I can see we're locking the host memory buffer on callbacks, but two threads could be locking different host memory buffer instances and thus still try to concurrently modify the registration map.

Reply:
I'll get answer for this soon. I think if it's a problem here, it's a problem for the device version of it (since I lifted the code from it). It may be good to try and generalize the code so I fix it one place.

Reply:
Sorry for the delay. This map is one-to-one with a cudf.HostColumnVector instance. The actions are all locked under this, and it is used in case two different RapidsHostColumnarBatch instances were to alias a particular cudf.HostColumnVector instance. So I don't think we need extra locking here.